tests/unit/acquisition/test_rule.py

```diff
@@ -186,3 +189,37 @@ def test_trust_region_for_unsuccessful_local_to_global_trust_region_reduced() ->
     npt.assert_array_less(current_state.eps, previous_state.eps)  # current TR smaller than previous
     assert current_state.is_global
     npt.assert_array_almost_equal(current_state.acquisition_space.lower, lower_bound)
+
+
+class _BatchModelMinusMeanMaximumSingleBuilder(BatchAcquisitionFunctionBuilder):
+    def prepare_acquisition_function(
+        self, dataset: Mapping[str, Dataset], model: Mapping[str, ProbabilisticModel]
+    ) -> BatchAcquisitionFunction:
+        return lambda at: -tf.reduce_max(model[OBJECTIVE].predict(at)[0], axis=-2)
```

Comment:
this is a little dense for a test. does this take the negative of the maximum predictive mean of the batch? does it assume the optimizer only queries one point `at` at a time?

Reply:
yes, that's what it returns.
the maximum predictive mean is the "worst" mean over the batch (when you minimize) so that maximising this acquisition will result with a batch where all points have a low predictive mean.
This works for at is [N, B, D]. Predict returns [N, B, 1] and we reduce along the -2 axis, which delivers [N, 1].

Reply:
i'm confused because you're indexing at 0, which I would have thought would index N

Reply:
[0] is to the get the mean only and not the variance. axis=-2 squashes the B dimension.

Reply:
ooooh, i missed that